ngapp/src/app/student/story-drawer/story-drawer.component.ts

```diff
@@ -121,4 +131,96 @@ export class StoryDrawerComponent implements OnInit {
       }
     });
   }
+
+  /**
+   * Open a dialog asking the user if they really want to delte their story
+   * Call the function to delete the story if the user clicks 'yes'
+   * @param id story id to be deleted
+   */
+  openDeleteStoryDialog(id: string) {
+    this.dialogRef = this.dialog.open(BasicDialogComponent, {
+      data: {
+        title: this.ts.l.delete_story,
+        message: "Are you sure you want to delete this story?",
+        confirmText: this.ts.l.yes,
+        cancelText: this.ts.l.no,
+      },
+      width: "50vh",
+    });
+
+    this.dialogRef.afterClosed().subscribe(async (res) => {
+      this.dialogRef = undefined;
+      if (res) {
+        this.deleteStory(id);
+      }
+    });
+  }
+
+  /**
+   * Delete the given story and any associated recordings
+   * Set the new 'current story' => next one in the list, or first one
+   * if the last story in the list was deleted
+   * @param id story id to be deleted
+   */
+  deleteStory(id: string) {
+    // remove any associated recordings
+    this.recordingService.deleteStoryRecordingAudio(id).subscribe((_) => {});
+    this.recordingService.deleteStoryRecording(id).subscribe((_) => {});
+
+    // get index of story to be deleted within story list
+    let storyIndex = this.stories.findIndex((story) => story._id === id);
+
+    // delete the story
+    this.storyService.deleteStory(id).subscribe((_) => {
+      this.engagement.addEventForLoggedInUser(EventType["DELETE-STORY"], { _id: id, });
+
+      // reset the story list to empty if list contains only one story
+      if (this.stories.length === 1) {
+        this.storyEmitter.emit(null);
+        this.stories = [];
+      }
+      // set the current story to the next story in the list if there is one
+      else if (storyIndex < this.stories.length - 1) {
+        let nextStory = this.stories[storyIndex + 1];
+        this.setStory(nextStory);
+        this.stories.splice(storyIndex, 1);
+      }
+      // set the first story as current story if the last story in list was deleted
+      else {
+        this.setStory(this.stories[0]);
+        this.stories.splice(storyIndex, 1);
+      }
```

Comment:
Maybe we could simplify this slightly be just _always_ setting the next story to the first in the list (i.e. the most recent story)? This way the behaviour is the same whether or not it's the `n-1`th story or any other.

```suggestion
      if (this.stores.length > 0) {
        this.setStory(this.stories[0]);
        this.stories.splice(storyIndex, 1);
      } else {
        this.storyEmitter.emit(null);
        this.stories = [];
      }
```